exotica/src/TaskMap.cpp

```diff
@@ -68,9 +68,36 @@ std::vector<KinematicFrameRequest> TaskMap::GetFrames()
     return Frames;
 }
 
-void TaskMap::taskSpaceDim(int& task_dim)
+void TaskMap::update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi, Eigen::MatrixXdRef J)
 {
-    task_dim = taskSpaceDim();
+    if (J.rows() != taskSpaceDim() && J.cols() != x.rows())
+        throw_named("Jacobian dimension mismatch!");
+
+    // Store original x (needs to be reset later).
+    Eigen::VectorXd x_original(x);
+    update(x_original, phi);
+
+    // Do not run finite differencing if phi is zero.
+    if (phi.isApprox(Eigen::VectorXd::Zero(taskSpaceDim())))
+    {
+        J.setZero();
+        return;
+    }
```

Comment:
This means that, e.g., if the end-effector is at its goal J=0. That is not true!